### PR TITLE
Add health summary api endpoint

### PR DESCRIPTION
**Pull request by fabriziosestito**
This PR adds the health summary API endpoint

The Health summary computation is wrapped in a service.
It also adds the possibility to filter by ID in the `GetAll` methods of cluster and hosts services.

### CODE REVIEW

File: web/sap_systems_api.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/trento-project/trento/web/services"
+)
+
+type HealthSummary struct {
+	ID              string `json:"id"`
+	SID             string `json:"sid"`
+	SAPSystemHealth string `json:"sap_system_health"`
```
Comment (nelsonkopliku) [inline web/sap_systems_api.go:13]: This is ignored, right?

**Review by nelsonkopliku — APPROVED** [review]
Great! :rocket:

**Review by nelsonkopliku — COMMENTED** [review]
When CI goes green :smile:

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hey @fabriziosestito ,
I think there is something wrong with the health computation. Comment below.
Could you check?

File: web/models/health_summary.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	HealthSummaryHealthUnknown  = "unknown"
+)
+
+type HealthSummary []SAPSystemHealthSummary
+
+type SAPSystemHealthSummary struct {
+	ID             string `json:"id"`
+	SID            string `json:"sid"`
+	ClustersHealth string `json:"clusters_health"`
+	DatabaseHealth string `json:"databases_health"`
```
Comment (arbulu89) [inline web/models/health_summary.go:16]: Maybe `DatabasesHealth` better (last `s` missing). Or remove the `s` from `databases_health` as we will only have 1 database per system

File: web/sap_systems_api.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+	"github.com/gin-gonic/gin"
+	"github.com/trento-project/trento/web/services"
+)
+
+// ApiSAPSystemsHealthSummaryHandler godoc
+// @Summary Retrieve SAP Systems Health Summary
+// @Accept json
+// @Produce json
+// @Success 200 {object} HealthSummaryResponse
```
Comment (arbulu89) [inline web/sap_systems_api.go:14]: Where is this `HealthSummaryResponse` defined? i cannot find it anywhere...

File: web/services/health_summary_service.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	case models.SAPSystemHealthWarning:
+		return models.HealthSummaryHealthWarning
+	case models.SAPSystemHealthCritical:
+		return models.HealthSummaryHealthCritical
+	default:
+		return models.HealthSummaryHealthUnknown
+	}
+}
+
+func computeAggregatedClustersHealth(clusters []*models.Cluster) string {
```
Comment (arbulu89) [inline web/services/health_summary_service.go:80]: As commented offline, hopefully we can remove all of this duplicated code at some point hehe

File: web/models/health_summary.go (hunk 2 above)
Comment (fabriziosestito) [reply web/models/health_summary.go]: i blame copilot, fixed, thanks!

File: web/sap_systems_api.go (hunk 3 above)
Comment (fabriziosestito) [reply web/sap_systems_api.go]: updated with the right struct, thanks!

**Pull request merged**